src/components/large-pipeline-warning/large-pipeline-warning.test.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import ConnectedModal, {
+  LargePipelineWarning,
+  mapStateToProps,
+  mapDispatchToProps,
+} from './index';
+import { mockState, setup } from '../../utils/state.mock';
+
+describe('large modal', () => {
+  it('renders without crashing', () => {
+    const mockFn = jest.fn();
+    const props = {
+      onToggleIgnoreLargeWarning: mockFn,
+      nodes: [],
+      visible: true,
+    };
+    const wrapper = setup.mount(<LargePipelineWarning {...props} />);
+    expect(wrapper.find('.pipeline-warning').length).toBe(1);
+  });
+
+  it('clicking the render anyways button will toggle the graph to display', () => {
+    const mockFn = jest.fn();
+    const props = {
+      onToggleIgnoreLargeWarning: mockFn,
+      nodes: [],
+      visible: true,
+    };
+    const wrapper = setup.mount(<LargePipelineWarning {...props} />);
+
+    wrapper.find('.kui-button__btn').simulate('click');
```

Comment:
Ideally could use a more specific class here like `pipeline-warning__render-button`?

Reply:
This class is part of Kedro UI so it can't be changed.

Reply:
@richardwestenra you mean that button component doesn't let you add an additional user defined class?

Reply:
there is no need for an additional user defined class for the button as the design of this button is exactly as is

Reply:
@bru5 Correct https://github.com/quantumblacklabs/kedro-ui/blob/master/src/components/button/button.jsx

And I agree with Susanna that it's not worth adding a new class just for the test.

Reply:
@richardwestenra okay seems like an upstream issue then, seems fine to leave it here